### PR TITLE
[zh] Sync access-authn-authz/authentication.md

### PR DESCRIPTION
**Pull request by Zhuzhenghao**
This pr 

1. updates links to Dex referring to upstream #25147
2. clarifies tokenreview API referring to upstream #24349
3. removes redundant blank lines

### CODE REVIEW

**Comment by tengqm** [comment]
Please check the output from:

```
./scripts/lsync.sh content/zh-cn/docs/reference/access-authn-authz/authentication.md
```

/hold

File: content/zh-cn/docs/reference/access-authn-authz/authentication.md [hunk 1]
```diff
@@ -495,26 +495,26 @@ sequenceDiagram
...
+8. Once authorized the API server returns a response to `kubectl`
+9. `kubectl` provides feedback to the user
 -->
-1.  登录到你的身份服务（Identity Provider）
-2.  你的身份服务将为你提供 `access_token`、`id_token` 和 `refresh_token`
-3.  在使用 `kubectl` 时，将 `id_token` 设置为 `--token` 标志值，或者将其直接添加到
+1. 登录到你的身份服务（Identity Provider）
+2. 你的身份服务将为你提供 `access_token`、`id_token` 和 `refresh_token`
+3. 在使用 `kubectl` 时，将 `id_token` 设置为 `--token` 标志值，或者将其直接添加到
     `kubeconfig` 中
```
Comment (tengqm) [inline content/zh-cn/docs/reference/access-authn-authz/authentication.md:511]:
```suggestion
   `kubeconfig` 中
```

File: content/zh-cn/docs/reference/access-authn-authz/authentication.md [hunk 2]
```diff
@@ -744,6 +746,9 @@ Webhook 身份认证是一种用来验证持有者令牌的回调机制。
   其中描述如何访问远程的 Webhook 服务。
 * `--authentication-token-webhook-cache-ttl` 用来设定身份认证决定的缓存时间。
   默认时长为 2 分钟。
+* `--authentication-token-webhook-version` 决定是使用 `authentication.k8s.io/v1beta1` 还是
+  `authenticationk8s.io/v1`。
```
Comment (tengqm) [inline content/zh-cn/docs/reference/access-authn-authz/authentication.md:750]:
```suggestion
  `authenticationk8s.io/v1` 版本的
```

**Comment by windsonsea** [comment]
Thanks.
/lgtm

**Pull request merged**